lib/res_man.rb

```diff
@@ -0,0 +1,113 @@
+module Stayappy
+  class ReservationManager
+    
+    attr_reader :rooms, :bookings, :block_reservations 
+    
+    STANDARD_ROOM_COST = 200
+
+    def initialize
+      @rooms = []
+      20.times do |room_num|
+        @rooms << Room.new(room_num: room_num + 1, cost: STANDARD_ROOM_COST)
+      end
+      @bookings = []
+      @block_reservations = []
+    end
+
+    def assign_room(check_in, check_out)
+      available_rooms = view_by_vacancy(check_in, check_out)
+      if available_rooms.length == 0
+        raise ArgumentError.new("No rooms available for reservation window #{check_in} to #{check_out}")
+      end
+      available_rooms[0]
+    end
+
+    def make_reservation(check_in, check_out)
+      room = assign_room(check_in, check_out)
+      reservation = Reservation.new(room, check_in, check_out)
+      @bookings << reservation
+      reservation
+    end
+
+    def reservations_by_room(room_num, start_date, end_date)
+      matching_reservations = []
+      @bookings.each do |reservation|
+        puts reservation.room.room_num
+        if (reservation.in_range?(start_date, end_date) && reservation.room.room_num == room_num)
+            matching_reservations << reservation
+        end
+      end
+      matching_reservations
+    end
+
+    def reservations_by_date(date)
+      matching_reservations = []
+      @bookings.each do |reservation|
+        puts reservation.room.room_num
+        if reservation.in_range?(date, date)
+            matching_reservations << reservation
+        end
+      end
+      matching_reservations
+    end
+
+    def view_by_vacancy(check_in, check_out)
+      available_rooms = @rooms.clone
+      # Loop through all the bookings to see which rooms are taken
+      @bookings.each do |reservation|
```

Comment:
Good work including regular booking and block reservations in your `view_by_vacancy` method.